solarkraft/test/unit/instrument.test.ts

```diff
@@ -141,7 +192,18 @@ describe('Apalache JSON instrumentor', () => {
     it('decodes Soroban Map values to Apalache JSON IR', () => {
         assertProperMap({})
         assertProperMap({ 1: 'a' })
-        assertProperMap({ a: 2, b: 3 })
+        assertProperMap({ 'a%': 2, b: 3 })
+        assertProperMap({ a: 2, _: 3 })
+    })
+
+    it('decodes Soroban struct values to Apalache JSON IR', () => {
+        assertProperStruct({ some_long_name: 1 })
+        assertProperStruct({ a: 2, b: 3 })
+    })
+
+    it('decodes Soroban enum values to Apalache JSON IR', () => {
+        assertProperEnum(['A', 1])
+        assertProperEnum(['B', 3, 44])
```

Comment:
Can you add some negative tests, i.e. expecting throws?

Reply:
You mean for the `assert(false)`?
It's literally meant to be unreachable, so not sure how you'd trigger it?

Reply:
No, I mean passing junk inputs. `tlaJsonOfNative` has an out-of-if return which makes `SetAsFun`. However, this should somehow break if it tries to construct e.g. `SetAsFun({ <<"2", 3>>, <<"4", "5">>})`, since that would be an illegal expression for apalache. Right now your test all check for "good" inputs.
or things like `assertProperEnum` asserting false when applied to e.g. a primitive int.

Reply:
I would say that we could write tests for the negative inputs earlier, but not now. We can make a separate issue on this.

Reply:
> Right now your test all check for "good" inputs.

They check for inputs that are possible from Soroban contracts. If you have an example of plausible input that isn't covered, please open an issue.